### PR TITLE
Add ProfileBundle and drop SonataUserBundle dependencies 

### PR DESCRIPTION
**Pull request by wbloszyk**
I havent time to write PR description. I will do it tomorrow.

This PR add profile which was drop after upgrade SonataUserBundle from 3.x => 4.x version

## Subject

I am targeting this branch, because {reason}.

Closes #{put_issue_number_here}

## Changelog

```markdown
### Added
- Added some `Class::newMethod` to do great stuff

### Changed

### Deprecated

### Removed

### Fixed

### Security
```

### CODE REVIEW

**Comment by greg0ire** [comment]
I just released 3.0.0 , hence the retarget. If you think there will be BC-breaks in your PR, let's think together how we can avoid them, and if we find no solution, let's target `master` again.

**Comment by wbloszyk** [comment]
I can change profile configuration to auto check sonatauserbundle version or leave it as e-commerce configuration option. Then this PR will be BC.

File: src/ProfileBundle/Block/AccountBlockService.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    private $tokenStorage;
+
+    /**
+     * NEXT_MAJOR: Go back to type hinting check when bumping requirements to SF 2.6+.
+     *
+     * @param string                                         $name
+     * @param TokenStorageInterface|SecurityContextInterface $tokenStorage
+     */
+    public function __construct($name, EngineInterface $templating, $tokenStorage)
+    {
```
Comment (jeroendk) [inline src/ProfileBundle/Block/AccountBlockService.php]: Maybe add the type hint? Like the comment mentions:
'NEXT_MAJOR: Go back to type hinting check when bumping requirements to SF 2.6+.'

Comment (wbloszyk) [reply src/ProfileBundle/Block/AccountBlockService.php]: I miss it. SecurityContextInterface should be deleted becouse ecommere require symfony 3.4 || 4.x

**Comment by wbloszyk** [comment]
I must fix my fork branches. I will add another PR.

**Pull request closed without merging**